pkgs/tools/admin/pgadmin/update.sh

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p curl wget jq yq common-updater-scripts prefetch-yarn-deps yarn-lock-converter
+
+set -eu -o pipefail
+
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+NC='\033[0m' # No Color
+
+################################################################
+#         This script will update pgadmin4 in nixpkgs          #
+# Due to recent changes upstream, we will need to convert the  #
+#             `yarn.lock` file back to version 1.              #
+#   This isn't trivially done and relies on 3rd party tools    #
+#       and a hand-written converter (in this script).         #
+# Also, the converter cannot check for `github` repos in the   #
+#   `yarn.lock` file, which this script will add automatically #
+################################################################
+
+
+
+##################################################
+# Check for new version and download #
+##################################################
+
+scriptDir=$(cd "${BASH_SOURCE[0]%/*}" && pwd)
+nixpkgs=$(realpath "$scriptDir"/../../../..)
+
+newest_version="$(curl -s https://www.pgadmin.org/versions.json | jq -r .pgadmin4.version)"
+old_version=$(nix-instantiate --eval -E "(import \"$nixpkgs\" { config = {}; overlays = []; }).pgadmin4.version" | tr -d '"')
+url="https://ftp.postgresql.org/pub/pgadmin/pgadmin4/v${newest_version}/source/pgadmin4-${newest_version}.tar.gz"
+
+if [[ $newest_version == $old_version ]]; then
+  printf "${GREEN}Already at latest version $newest_version${NC}\n"
+  exit 0
+fi
+printf "${YELLOW}New version: $newest_version ${NC}\n"
+
+# don't use mktemp, so if a network error happens, we can resume from there
+mkdir -p /tmp/pgadmin-update-script
+pushd /tmp/pgadmin-update-script
+wget -nc $url
+tar -xzf "pgadmin4-$newest_version.tar.gz"
+cd "pgadmin4-$newest_version/web"
+
```

Comment:
We want to use mktemp to make the directory not predictable and don't interfere with previous runs of the script. This is also missing a trap to clean the directory up in case of an error

Reply:
I specifically didn't want to use `mktemp` for this exact reason I stated in the comment.
The `yarn-lock-converter` failed  on two test runs midway and this makes the tool resume. Also it is located in `/tmp` which gets wiped on reboot anyway. 
In my opinion this should _specifically_  not use `mktemp`

Reply:
> Also it is located in `/tmp` which gets wiped on reboot anyway.

That's not true https://search.nixos.org/options?channel=22.11&show=boot.cleanTmpDir&from=0&size=50&sort=relevance&type=packages&query=cleanTmp

I had my /tmp multiple times fill up because of faulty update scripts which usually breaks a lot of things

Reply:
Good point. The script will always ask to delete the temp folder when terminated prematurely, so it shouldn't fill up `/tmp` 
OTOH it has the option to _not_ do that, so downloads can be resumed